components/zcash_address/src/kind/unified/fvk.rs

```diff
@@ -0,0 +1,360 @@
+use std::cmp;
+use std::convert::{TryFrom, TryInto};
+
+use super::{
+    private::{SealedContainer, SealedItem},
+    Container, Encoding, ParseError, Typecode,
+};
+
+/// The set of known FVKs for Unified FVKs.
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub enum Fvk {
+    /// The raw encoding of an Orchard Full Viewing Key.
+    ///
+    /// `(ak, nk, rivk)` each 32 bytes.
+    Orchard([u8; 96]),
+
+    /// Data contained within the Sapling component of a Unified Full Viewing Key
+    ///
+    /// `(ak, nk, ovk, dk)` each 32 bytes.
+    Sapling([u8; 128]),
+
+    /// A pruned version of the extended public key for the BIP 44 account corresponding to the
+    /// transparent address subtree from which transparent addresses are derived. This
+    /// includes just the chain code (32 bytes) and the public key (33 bytes) and excludes
```

Comment:
```suggestion
    /// includes just the chain code (32 bytes) and the compressed public key (33 bytes), and excludes
```
We should be explicit about this, because a secp256k1 uncompressed public key encoding is also 65 bytes.